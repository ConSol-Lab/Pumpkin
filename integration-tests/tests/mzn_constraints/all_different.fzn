array [1..4] of var 1..4: xs :: output_array(1..4);

constraint pumpkin_all_different(xs);

solve satisfy;
