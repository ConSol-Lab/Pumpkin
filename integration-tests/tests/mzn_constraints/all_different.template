predicate fzn_all_different_int(array [int] of var int: vars);

array [1..4] of var 1..4: xs :: output_array([1..4]);

constraint fzn_all_different_int(xs);

solve satisfy;
