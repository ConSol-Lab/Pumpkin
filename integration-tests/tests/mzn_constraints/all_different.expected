xs = array1d(1..4,[4, 3, 2, 1]);
----------
xs = array1d(1..4,[3, 4, 2, 1]);
----------
xs = array1d(1..4,[4, 2, 3, 1]);
----------
xs = array1d(1..4,[2, 4, 3, 1]);
----------
xs = array1d(1..4,[3, 2, 4, 1]);
----------
xs = array1d(1..4,[2, 3, 4, 1]);
----------
xs = array1d(1..4,[4, 3, 1, 2]);
----------
xs = array1d(1..4,[3, 4, 1, 2]);
----------
xs = array1d(1..4,[4, 1, 3, 2]);
----------
xs = array1d(1..4,[1, 4, 3, 2]);
----------
xs = array1d(1..4,[3, 1, 4, 2]);
----------
xs = array1d(1..4,[1, 3, 4, 2]);
----------
xs = array1d(1..4,[4, 2, 1, 3]);
----------
xs = array1d(1..4,[2, 4, 1, 3]);
----------
xs = array1d(1..4,[4, 1, 2, 3]);
----------
xs = array1d(1..4,[1, 4, 2, 3]);
----------
xs = array1d(1..4,[2, 1, 4, 3]);
----------
xs = array1d(1..4,[1, 2, 4, 3]);
----------
xs = array1d(1..4,[3, 2, 1, 4]);
----------
xs = array1d(1..4,[2, 3, 1, 4]);
----------
xs = array1d(1..4,[3, 1, 2, 4]);
----------
xs = array1d(1..4,[1, 3, 2, 4]);
----------
xs = array1d(1..4,[2, 1, 3, 4]);
----------
xs = array1d(1..4,[1, 2, 3, 4]);
----------
==========
